### ISSUE TITLE
Field flow: Have option to suppress required visuals where everything in the form is required

### ISSUE THREAD

**Issue by pjackson28** [wet-boew/GCWeb#1391]
Have a request from a client to suppress the required visuals (red asterisk and (required) at the end of the label) for required fields in a form where all fields are required (so keep the fields validated as required, just avoid the repetitive visuals).

The client is questioning why Field flow forces this since it is not a restriction for the WET Form validation plugin. With the Form validation plugin, you just need to remove the ``required`` class from the label (leaving the ``required`` class on the field itself) and then drop the ``(required)`` text from the label. This of course should only be used for cases where all fields are optional or mandatory (the visuals will continue to be needed when there is a mix of required and optional).

It is only an issue for Field flow because that plugin generates the markup for the label, automatically applying the ``required`` class and the ``(required)`` text without providing a means of disabling this (other than providing an option for making the field completely optional, which is not what is wanted since the fields themselves should still be required).

**Comment by pjackson28** [comment]
From what I recall, control over the required visuals was part of the original template (through markup). That is part of why we went with the required class on the label and inserted required text in the label for the required visuals (so required aspect of the field and the required visuals were kept separate).

Wouldn't it be easier just to add an option that separates isOptional from the visuals? For instance you could just add something like a boolean displayRequired which defaults to true if isOptional is false (and is ignored if isOptional is true).

**Comment by duboisp** [comment]
Yes I can look of a quick solution that would add an new option like "displayRequired".
